nautobot_ssot/integrations/servicenow/models.py

```diff
@@ -0,0 +1,42 @@
+"""Configuration data model for nautobot-ssot-servicenow."""
+
+from django.db import models
+from django.shortcuts import reverse
+
+from nautobot.core.models import BaseModel
+
+
+class SSOTServiceNowConfig(BaseModel):
+    """Singleton data model describing the configuration of this plugin."""
+
+    def delete(self, *args, **kwargs):
+        """Cannot be deleted."""
+
+    @classmethod
+    def load(cls):
+        """Singleton instance getter."""
+        if cls.objects.all().exists():
+            return cls.objects.first()
+        return cls.objects.create()
+
+    servicenow_instance = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text="ServiceNow instance name, will be used as <code>&lt;instance&gt;.servicenow.com</code>.",
+    )
+
+    servicenow_secrets = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="Secrets group for authentication to ServiceNow. Should contain a REST username and REST password.",
+    )
```

Comment:
I don't see any documentation in this PR. Seems like at the very least the configuration of this integration should be documented, such as how/when to use the nautobot_config.py configuration vs the model-driven configuration?

Reply:
Documentation will be done in separate PR. Similar to ChatOps integrations.